File: 프로그래머스/2/155651. 호텔 대실/호텔 대실.js
function makeMin(time) {
    const [hour, min] = time.split(":").map(v => Number(v));
    return hour * 60 + min;
}

function solution(book_time) {
    const timeArr = Array.from({ length: makeMin('23:59') + 10 }, () => 0);

    book_time.forEach((time, i) => {
        const [s, e] = time;
        let start = makeMin(s);
        const end = makeMin(e) + 9;

        for (start; start <= end; start++) timeArr[start]++;
    });

    return Math.max(...timeArr);
}

File: 프로그래머스/2/155651. 호텔 대실/호텔 대실.py
def solution(book_time):
    bookings = [(int(start[:2]) * 60 + int(start[3:]), 
                 int(end[:2]) * 60 + int(end[3:]) + 10) 
                for start, end in book_time]
    bookings.sort()
    
    rooms = []
    
    for start, end in bookings:
        room_found = False
        for i in range(len(rooms)):
            if rooms[i] <= start:
                rooms[i] = end
                room_found = True
                break
        
        if not room_found: rooms.append(end)
    
    return len(rooms)